### PR TITLE
Upgrade app UI and address old issues

### PR DESCRIPTION
**Pull request by nvarner**
# Upgrade app UI and address old issues

This PR updates the app's UI to be more useful and solve several old issues.

Changes:
- Create cards for user utterances (#49, some of #50)
- Remove "Mycroft says:" (#49)
- Long press cards for options (#53)
- Change the TTS toggle switch to a button
- Remove option to toggle the TTS toggle
- Create a keyboard mode toggle to allow for keyboard and voice input

### Microphone mode
screenshot_2019-02-18-13-35-42

### Keyboard mode
screenshot_2019-02-18-13-36-02

### Keyboard input is sent to Mycroft
screenshot_2019-02-18-13-36-10

### Long press on Mycroft utterance
screenshot_2019-02-18-13-36-20

### Long press on user utterance
screenshot_2019-02-18-13-36-15

### CODE REVIEW

File: mobile/src/main/java/mycroft/ai/MainActivity.kt [hunk 1]
```diff
@@ -95,8 +91,42 @@ class MainActivity : AppCompatActivity() {
...
+                // Switch to keyboard
+                micButton.visibility = View.INVISIBLE
+                utteranceInput.visibility = View.VISIBLE
+                sendUtterance.visibility = View.VISIBLE
+            }
+        }
+
+        micButton.setOnClickListener { promptSpeechInput() }
+        sendUtterance.setOnClickListener {
+            val utterance = utteranceInput.text.toString()
```
Comment (skraynick) [inline mobile/src/main/java/mycroft/ai/MainActivity.kt:122]: Why create variable. Think you can just use uteranceInput.... with out that. As the variable is just used once.

File: mobile/src/main/java/mycroft/ai/MainActivity.kt [hunk 2]
```diff
@@ -149,6 +179,32 @@ class MainActivity : AppCompatActivity() {
...
     }
 
+    override fun onContextItemSelected(item: MenuItem): Boolean {
+        super.onContextItemSelected(item)
+        if (item.itemId == R.id.user_resend) {
+            // Resend user utterance
+            sendMessage(utterances[currentItemPosition].utterance)
+        } else if (item.itemId == R.id.user_copy || item.itemId == R.id.mycroft_copy) {
+            // Copy utterance to clipboard
+            val cb = getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
```
Comment (skraynick) [inline mobile/src/main/java/mycroft/ai/MainActivity.kt]: I like good variable names. Can we make this something more descriptive...

**Review by skraynick — COMMENTED** [review]
Please check individual comments. thanks

File: mobile/src/main/java/mycroft/ai/MainActivity.kt (hunk 1 above)
Comment (nvarner) [reply mobile/src/main/java/mycroft/ai/MainActivity.kt:122]: I was going to use that variable on the next line (`sendMessage(utteranceInput.text.toString())` as `sendMessage(utterance)`) but forgot to make the change. Would that be an appropriate fix or do you still want me to remove the variable?

**Comment by nvarner** [comment]
I changed line 123 to use `utterance` and renamed `cb` to `clipboardManager`.

**Comment by testman42** [comment]
I would just like to point out that Chuck Norris ~~joke~~ **fact** in screenshots above is very funny :+1:

**Pull request merged**